guides/javascript.md

```diff
@@ -0,0 +1,264 @@
+✅
+
+``` javascript
+function formatChangelog() {
+  // ...
+}
+```
+
+For a function or method that returns a boolean, prefix the name with a descriptive verb so that readers don't confuse it with a boolean variable.
+
+🚫
+
+``` javascript
+class NetworkController {
+  isEIP1559Compatible() {
+    // ...
+  }
+}
+```
+
+✅
+
+``` javascript
+class NetworkController {
+  determineEIP1559Compatibility() {
+    // ...
+  }
+}
+```
+
+### Read more
+
+- [Google's JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html#naming-method-names)
+- [.NET Fundamentals](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods)
+- [Kotlin Coding Conventions](https://kotlinlang.org/docs/coding-conventions.html#choose-good-names)
+- [Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage)
+
+## Use a verb phrase for boolean variable names that do not involve secondary objects
+
+Sometimes a variable that is intended to hold a boolean value does not have an explicit subject, but represents the context where the variable is defined (e.g., a class or an entire file). When naming such a variable, use a statement that describes the state of the subject, minus the name of the subject itself. Usually this means prefixing the name with a form of "to be" or "to have" (e.g. `is*`, `has*`), but you may find it more readable to use past or future tense and/or a modal verb such as `should`.
+
+🚫
+
+``` javascript
+const removed = false;
+```
+
+✅
+
+``` javascript
+// Any of these would do
+const isRemoved = false;
+const wasRemoved = false;
+const hasBeenRemoved = false;
+const shouldBeRemoved = false;
+```
+
+If the name represents a negative statement, reword it into a positive statement and inverting the value assigned to the value.
+
+🚫
+
+``` javascript
+const notEnoughGas = false;
+```
+
+✅
+
+``` javascript
+const hasEnoughGas = true;
+```
+
+Take special note of variables which are created via React's `useState` hook.
+
+🚫
+
+``` javascript
+const [removed, setRemoved] = useState(false);
+```
+
+✅
+
+``` javascript
+const [isRemoved, setIsRemoved] = useState(false);
+```
+
+## 💡 Place names of secondary concepts first in boolean variable names
+
+Sometimes a boolean variable describes a subject that's different than the context where the variable is defined:
+
+``` javascript
+const ownedAccount = Boolean(ownedAccountName);
+```
+
+In naming such a variable, there are a couple of different routes you could take.
+
+One way is to always place the verb at the beginning:
+
+``` javascript
+const isRecipientOwnedAccount = Boolean(ownedAccountName);
+```
+
+However, this naming strategy creates a point of friction for objects, arrays, or React components, where it may be desirable to sort identifiers alphabetically. In that case you could end up with something like:
+
+``` jsx
+<SenderToRecipient
+  isRecipientOwnedAccount={isRecipientOwnedAccount}
```

Comment:
This example might be improved if we could think of a prop to go in-between this one as the `recipient__` ones, Just to make super clear what the advantages are of this naming convention.

Reply:
onClick maybe?

Reply:
Makes sense, updated here: [7b8e6ae](https://github.com/MetaMask/contributor-docs/pull/11/commits/7b8e6ae0de276d38b6e58adae5a97d3140738ed9)